### DISCUSSION TITLE
파동과 FSK에 대한 이해

### DISCUSSION THREAD

**Discussion by zion830** [euphony-io/euphony#190] [Knowledge Share]
저희 1팀 분들과 함께 **ASK, FSK, PSK, PCM**을 스터디한 내용입니다! 그 중에 `FSK` 부분을 업로드합니다 🙂

# FSK의 정의?
한국 버전 위키에 나와 있는 정의입니다.

> 주파수 편이 변조(Frequency-shift keying, FSK)란 반송파(carrier wave)의 이산 주파수(discrete frequencies) 변화를 통해 디지털 정보가 전송되는 주파수 변조 체계이다

영문 버전 위키에 나와있는 정의입니다.
> Frequency-shift keying (FSK) is a frequency modulation scheme in which digital information is transmitted through discrete frequency changes of a carrier signal.

수능 비문학 같습니다. 읽을 수 있는데 읽을 수가 없습니다.
반송파, 주파수, 변조... 문장에 들어간 단어를 먼저 해석해 보겠습니다.

# 사전 지식
### Signal (신호)
신호는 데이터를 전송하기 위해 사용되는 파형(waveform)입니다.
신호는 크게 두가지가 존재합니다. `디지털 신호`와 `아날로그 신호`입니다.
* 디지털 신호는 0과 1로 이루어져 있습니다.
* 아날로그 신호는 온도, 길이, 각도 처럼 현실 세계에서 발생하는 정보로, 시간이 시간이 지남에 연속적으로 변화합니다.

### 파동(wave)의 구성 요소들
먼저 파동(wave)의 진폭, 주파수, 위상에 대해 간단히 소개합니다.
* 진폭(amplitude) : 아래와 같이 파동을 나타내는 그래프가 있을때, 중간인 0에서부터 최고점을 수직으로 연결했을 때의 높이가 진폭입니다. 공기 분자가 얼마나 강하게 진동했느냐로 진폭이 정해지기 때문에, 소리가 클수록 진폭이 커지고, 파동은 강해집니다. 작게 속삭일 때 발생하는 진폭보다 크게 소리칠 때의 진폭이 더 큽니다.
![Digital modulation_ ASK, FSK, and PSK 0-54 screenshot](https://user-images.githubusercontent.com/35446215/185762650-49278e44-7a59-4603-b90d-7dab804254c8.png)
* 파장(wavelength) : 파동의 형태가 주기적으로 반복될 때, 그 반복되는 구간의 길이를 나타냅니다. wave cycle의 최고점에서 다음 최고점까지의 가로 길이를 측정합니다.
![image](https://user-images.githubusercontent.com/35446215/185779837-af8a01c4-fb5d-4b8a-a2b7-c871528b7dea.png)

* 주파수(frequency) : 우리가 흔히 '빈도'로 알고 있는 단어인데요, 주파수 또는 진동수 라고도 부릅니다. 1초간 발생한 진동 횟수를 의미합니다. 즉, 1초 사이에 발생한 wave cycle의 갯수입니다. 표시하는 단위는 Hz입니다.
* 위상(phase) : 위상은 wave cycle에서 특정한 지점(point)이 어느 단계에 위치하는가를 나타냅니다. 이때 이 위치는 각도로 표현합니다. 두 지점의 위상차는, 이 둘이 서로 얼마나 다른 단계에 있는지를 나타냅니다. 
![Digital modulation_ ASK, FSK, and PSK 1-49 screenshot](https://user-images.githubusercontent.com/35446215/185762857-babe6e8d-9e17-4052-8102-48208e6ece85.png)

### Modulation (변조)
변조는 **전송에 적합한 형태로 신호를 변환하는 과정**을 의미합니다. 전송에 적합한 형태로 데이터를 싣게 될 더 높은 주파수의 음파를 `반송파(carrier signal)`이라 합니다. 신호를 있는 그대로 전송할 수도 있을 텐데, 왜 변조가 필요한 걸까요?

**변조가 필요한 이유**
신호를 송수신하기 위해 안테나라는 장치를 사용합니다. 안테나 길이는 신호의 크기에 의해 결정됩니다. 신호의 크기가 큰 고주파 대역일수록 안테나의 길이는 짧아집니다.

변조를 거치지 않은 원본 신호를 `기저대역 신호(baseband signal)`라 합니다. 기저대역 신호를 있는 그대로 전송할 경우, 주파수가 너무 낮아 문제가 발생합니다. 낮은 주파수 때문에 먼 거리를 이동할 수 없고, 아주 큰 안테나가 필요해집니다. 

따라서 기저대역 신호를 파장이 짧은 고주파의 신호로 변환하기 위해 변조가 필요합니다.

# FSK

아날로그 신호를 변조하는 것을 `아날로그 변조`, 디지털 신호를 변조하는 것을 `디지털 변조`라 합니다. 
ASK와 FSK는 디지털 변조의 일종입니다. 디지털 변조를 거치고 나면, 기존의 데이터는 반송파에 실려 아날로그 신호가 됩니다. ASK와 FSK는 변조 과정에서 반송파의 어떤 부분을 변형시키느냐에서 차이가 갈립니다. 

ASK는 `amplitude`를 포함하고 있으니 진폭을 변형한다는 것, 그리고 FSK는 `frequency`를 포함하고 있으니 주파수를 변형할 것 같다는 느낌적인 느낌이 듭니다.
![Digital modulation_ ASK, FSK, and PSK 3-13 screenshot](https://user-images.githubusercontent.com/35446215/185764252-662406bf-af63-4958-a16d-f344e2c2cde1.png)

`BFSK(Binary FSK)`는 가장 간단하고, 흔한 형태의 FSK 구현입니다. 입력 신호는 0과 1 딱 2가지의 값만을 가지고 있고, 이 정보를 전송하기 위해 한쌍의 이산 주파수를 사용합니다.

위에 나온 그림처럼, FSK는 2개의 주파수로 이진 데이터를 표현할 수 있습니다. 디지털 신호가 1일 때, 주파수는 더 높아집니다. 디지털 신호가 0일때 주파수는 반대로 낮아집니다.

FSK의 특징은 다음과 같습니다.
* 진폭이 일정하고 주파수로 정보를 담기 때문에 Noise에 강합니다. 그래서 데이터가 오류 없이 수신될 확률이 높습니다.
* decode가 쉽습니다.
* 전력 요구량이 일정합니다.
* 주파수가 계속 바뀌면서, 불연속적인 지점이 생기기 때문에 순간적으로 대역폭이 넓어지는 문제가 있습니다. 이로인해 대역폭 효율성이 떨어집니다.
* 빠른 속도가 필요하거나, 대용량 통신에서는 사용이 어렵습니다. 

## Reference
* https://unacademy.com/content/neet-ug/study-material/physics/relation-between-phase-and-frequency/
* https://www.youtube.com/watch?v=qGwUOvErR8Q
* https://www.youtube.com/watch?v=mHvV_Tv8HDQ
* https://www.scienceall.com/%ED%8C%8C%EC%9E%A5wave-length-2/
* https://www.carritech.com/news/what-is-modulation-in-telecommunications/
* https://www.techopedia.com/definition/3004/frequency-shift-keying-fsk
* https://ko.wikipedia.org/wiki/%EB%B3%80%EC%A1%B0
* https://ko.wikipedia.org/wiki/%EC%A3%BC%ED%8C%8C%EC%88%98_%ED%8E%B8%EC%9D%B4_%EB%B3%80%EC%A1%B0
* https://www.ecstuff4u.com/2018/08/advantages-and-disadvantages-of-fsk.html
* https://www.niu.edu/remote-lab/resources/graphical-user-interfaces/frequency-shift.shtml

**Comment by zion830** [comment]
FSK 코드 상으로의 구현은 작년에 다른 멘티분이 지식 공유해주신 내용이 있어서 링크 걸어둡니다! #115

**Comment by designe** [comment]
FSK 지식공유 너무 좋습니다 :)
FSK 진보된 CPFSK 라는 친구도 있어요!
Continuous Phase Frequency Shift Keying이라는 친구입니다.
FSK 진화형으로 볼 수 있어요!
